control-plane/config-entries/controllersv2/meshconfig_controller_test.go

```diff
@@ -145,6 +145,14 @@ func TestMeshConfigController_createsMeshConfig(t *testing.T) {
 			resourceClient, err := consul.NewResourceServiceClient(testClient.Watcher)
 			require.NoError(t, err)
 
+			require.Eventually(t, func() bool {
+				_, _, err := testClient.APIClient.Partitions().Read(context.Background(), constants.DefaultConsulPartition, nil)
+				if err != nil {
+					return false
+				}
+				return true
+			}, 5*time.Second, 500*time.Millisecond)
+
```

Comment:
Ideally, this moves to Consul's `sdk`, but it needs to be done in a way that works across V1 and V2 that will require some more thought, more than cleaning up test failure on `main`.